src/rebuild.ts

```diff
@@ -184,19 +188,40 @@ class Rebuilder {
   }
 
   rebuildAllModulesIn(nodeModulesPath: string, prefix = '') {
-    d('scanning:', nodeModulesPath);
+    // While we use `cnpm`, it will make a circle scanning the dep tree.
+    // We also need to ensure that the `node_modules` which we are scanning has never came before.
+    const realNodeModulesPath = fs.realpathSync(nodeModulesPath);
+    if (this.realNodeModulesPaths.indexOf(realNodeModulesPath) > -1) {
+      return;
+    } else {
+      this.realNodeModulesPaths.push(realNodeModulesPath);
+    }
+
+    d('scanning:', realNodeModulesPath);
+
+    for (const modulePath of fs.readdirSync(realNodeModulesPath)) {
+      // If we use `cnpm` to install mudules, it will be rebuilded failed,
+      // because of some same modules rebuilded twice or more.
+      // Need to ensure one dep only be rebuilded once.
+      const finalPath = path.resolve(nodeModulesPath, modulePath);
+      const realPath = fs.realpathSync(finalPath);
+
+      if (this.realModulePaths.indexOf(realPath) > -1) {
+        continue;
+      } else {
```

Comment:
Again no need for the else block as their is a `continue` statement in the main if block